addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -402,11 +402,35 @@ def _get_rtc_server_info(self, rtc_session, ice_servers=None, key=None):
         json_web_token = jwt.sign(claims, key=key, ttl=60 * 60 * 8, algorithm=jwt.Algorithm.HS256)  # 8 hours
         return {"url": sfu_server_url, "channelUUID": sfu_channel_uuid, "jsonWebToken": json_web_token}
 
-    def _rtc_leave_call(self):
+    def _rtc_leave_call(self, recover=False):
         self.ensure_one()
         if self.rtc_session_ids:
             self.rtc_session_ids.unlink()
-        else:
+            if recover and self.partner_id:
```

Comment:
we could inverse this condition and return early if it's not met. This way, we would have less indentation for the lines below and it would be easier to read.